sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -1724,38 +1694,13 @@ static constexpr bfn_t operator^(bfn_t x, bfn_t y) {
 /// @param s1 Second boolean function argument.
 /// @param s2 Third boolean function argument.
 template <bfn_t FuncControl, typename T, int N>
-__ESIMD_API std::enable_if_t<std::is_integral_v<T>, __ESIMD_NS::simd<T, N>>
-bfn(__ESIMD_NS::simd<T, N> src0, __ESIMD_NS::simd<T, N> src1,
+__SYCL_DEPRECATED(
+    "Please use sycl::ext::intel::esimd::bfn<FuncControl>(src0, src1, src2);")
+__ESIMD_API std::enable_if_t<std::is_integral_v<T>, __ESIMD_NS::simd<T, N>> bfn(
+    __ESIMD_NS::simd<T, N> src0, __ESIMD_NS::simd<T, N> src1,
     __ESIMD_NS::simd<T, N> src2) {
-  if constexpr ((sizeof(T) == 8) || ((sizeof(T) == 1) && (N % 4 == 0)) ||
-                ((sizeof(T) == 2) && (N % 2 == 0))) {
-    // Bitcast Nx8-byte vectors to 2xN vectors of 4-byte integers.
-    // Bitcast Nx1-byte vectors to N/4 vectors of 4-byte integers.
-    // Bitcast Nx2-byte vectors to N/2 vectors of 4-byte integers.
-    auto Result = __ESIMD_ENS::bfn<FuncControl>(
-        src0.template bit_cast_view<int32_t>().read(),
-        src1.template bit_cast_view<int32_t>().read(),
-        src2.template bit_cast_view<int32_t>().read());
-    return Result.template bit_cast_view<T>();
-  } else if constexpr (sizeof(T) == 2 || sizeof(T) == 4) {
-    constexpr uint8_t FC = static_cast<uint8_t>(FuncControl);
-    return __esimd_bfn<FC, T, N>(src0.data(), src1.data(), src2.data());
-  } else if constexpr (N % 2 == 0) {
-    // Bitcast Nx1-byte vectors (N is even) to N/2 vectors of 2-byte integers.
-    auto Result = __ESIMD_ENS::bfn<FuncControl>(
-        src0.template bit_cast_view<int16_t>().read(),
-        src1.template bit_cast_view<int16_t>().read(),
-        src2.template bit_cast_view<int16_t>().read());
-    return Result.template bit_cast_view<T>();
-  } else {
-    // Odd number of 1-byte elements.
-    __ESIMD_NS::simd<T, N + 1> Src0, Src1, Src2;
-    Src0.template select<N, 1>() = src0;
-    Src1.template select<N, 1>() = src1;
-    Src2.template select<N, 1>() = src2;
-    auto Result = __ESIMD_ENS::bfn<FuncControl>(Src0, Src1, Src2);
-    return Result.template select<N, 1>();
-  }
+  return __ESIMD_NS::bfn<static_cast<__ESIMD_NS::bfn_t>(FuncControl)>(
+      src0, src1, src2);
```

Comment:
This static_cast is not needed after added that 'using' statement.
```suggestion
  return __ESIMD_NS::bfn<FuncControl>(src0, src1, src2);
```

Reply:
Right, thanks.